compiler/lib-wasm/generate.ml

```diff
@@ -1225,9 +1258,36 @@ let fix_switch_branches p =
       l
   in
   Addr.Map.iter
-    (fun _ block ->
+    (fun pc block ->
       match block.branch with
-      | Switch (_, l) -> fix_branches l
+      | Switch (x, l) -> (
+          match find_outlier_index l with
+          | `All_equals ->
```

Comment:
This optimization should already be handled by `Eval.eval_branch`. Maybe the `Distinguished` case should be performed there as well ..

Reply:
`Eval.eval_branch` only deals with the case where `x` is known.

Reply:
I moved it earlier. jsoo already optimizes switches. So, this does not make much of a difference.  The code is simpler, though.